modules/build/src/main/scala/scala/build/bsp/BspReloadableOptions.scala

```diff
@@ -0,0 +1,26 @@
+package scala.build.bsp
+
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.build.Logger
+import scala.build.blooprifle.BloopRifleConfig
+import scala.build.options.BuildOptions
+
+case class BspReloadableOptions(
+  buildOptions: BuildOptions,
+  bloopRifleConfig: BloopRifleConfig,
+  logger: Logger,
+  verbosity: Int
+)
+
+object BspReloadableOptions {
+  case class Reference(getReloaded: () => BspReloadableOptions) {
+    private val ref: AtomicReference[BspReloadableOptions] = new AtomicReference(getReloaded())
```

Comment:
I'm not sure we need to use an `AtomicReference` here, as we don't rely on its `compareAndSet` method. A simple private var would do (reading / writing a single reference var is thread safe).

Reply:
(This also applies to `atomicIntelliJ: AtomicBoolean` introduced in an earlier PR. A simple private var would have worked.)

Reply:
These private vars can be annotated with `volatile`.

Reply:
Huh, [TIL](https://stackoverflow.com/questions/281132/java-volatile-reference-vs-atomicreference), indeed.
You're right, fixed (including `BspServer.atomicIntelliJ` from a previous PR you mentioned).